packages/sign_in_with_apple/lib/src/authorization_credential.dart

```diff
@@ -1,34 +1,69 @@
+/// from your backend. Upon successful validation, you should create a session in your system for the current user,
+/// or consider her now logged in.
 @immutable
-abstract class AuthorizationCredential {}
-
-class AuthorizationCredentialAppleID implements AuthorizationCredential {
-  AuthorizationCredentialAppleID({
+class AuthorizationCredentialAppleID {
+  /// Creates an instance which contains the result of a successful Sign in with Apple flow.
+  const AuthorizationCredentialAppleID({
     @required this.userIdentifier,
     @required this.givenName,
     @required this.familyName,
     @required this.email,
-    @required this.identityToken,
     @required this.authorizationCode,
   }) : assert(authorizationCode != null);
 
+  /// An identifier associated with the authenticated user.
+  ///
+  /// This will be provided upon every sign-in.
+  /// This will stay the same between sign ins, until the user deauthorizes your App.
+  ///
   /// Can be `null`
   final String userIdentifier;
 
-  /// Can be `null`, will only be returned on the first authorization
+  /// The users given name, in case it was requested.
+  /// You will need to provide the [AppleIDAuthorizationScopes.fullName] scope to the [AppleIDAuthorizationRequest] for requesting this information.
+  ///
+  /// This information will only be provided on the first authorizations.
+  /// Upon further authorizations, you will only get the [userIdentifier],
+  /// meaning you will need to store this data securely on your servers.
+  /// For more information see: https://forums.developer.apple.com/thread/121496
+  ///
+  /// Can be `null`
   final String givenName;
 
-  /// Can be `null`, will only be returned on the first authorization
+  /// The users family name, in case it was requested.
+  /// You will need to provide the [AppleIDAuthorizationScopes.fullName] scope to the [AppleIDAuthorizationRequest] for requesting this information.
+  ///
+  /// This information will only be provided on the first authorizations.
+  /// Upon further authorizations, you will only get the [userIdentifier],
+  /// meaning you will need to store this data securely on your servers.
+  /// For more information see: https://forums.developer.apple.com/thread/121496
+  ///
+  /// Can be `null`
   final String familyName;
 
-  /// Can be `null`, will only be returned on the first authorization
-  final String email;
-
+  /// The users email in case it was requested.
+  /// You will need to provide the [AppleIDAuthorizationScopes.email] scope to the [AppleIDAuthorizationRequest] for requesting this information.
+  ///
+  /// This information will only be provided on the first authorizations.
+  /// Upon further authorizations, you will only get the [userIdentifier],
+  /// meaning you will need to store this data securely on your servers.
+  /// For more information see: https://forums.developer.apple.com/thread/121496
+  ///
   /// Can be `null`
-  final String identityToken;
```

Comment:
@tp What happened to the identity token? Is this not needed anymore?

Reply:
At the moment it's not supported. It would be the JWT token received from the web auth flow, but at AY we don't have any use for it (as we do the server-side validation).

Because of that we also have no good way of testing it. I don't know if this would be useful in any case to use instead of the server-side validated authorization code, but for now I thought it's better to drop it.